### PR TITLE
Add equalization of epochs randomly

### PR DESCRIPTION
**Pull request by mscheltienne**
Requested by a colleague in Geneva for sleep study epochs.
Adds the argument `'random'` to `mne.epochs.equalize_epoch_counts` to randomly select epochs.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -3837,16 +3837,17 @@ def equalize_epoch_counts(epochs_list, method="mintime"):
...
+        minimized. If ``'random'``, events will be randomly selected from each event
+        list.
 
     Notes
     -----
-    This tries to make the remaining epochs occurring as close as possible in
-    time. This method works based on the idea that if there happened to be some
-    time-varying (like on the scale of minutes) noise characteristics during
+    The method ``'mintime'`` tries to make the remaining epochs occurring as close as
+    possible in time. This method works based on the idea that if there happened to be
```
Comment (drammock) [inline mne/epochs.py:3849]: I think "works based on the idea that" is not quite right here ... Maybe "is motivated by the possibility that" would be better?

**Review by drammock — CHANGES_REQUESTED** [review]

File: mne/epochs.py (hunk 1 above)
Comment (mscheltienne) [reply mne/epochs.py]: Done

**Review by drammock — APPROVED** [review]
thanks for the quick fixes!

**Comment by mscheltienne** [comment]
Failures of the `pip-pre` workflow due to the dependency installation step, unrelated to this PR.

File: mne/utils/docs.py [hunk 2]
```diff
@@ -1289,6 +1289,14 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
     time are included. Defaults to ``-0.2`` and ``0.5``, respectively.
 """
 
+docdict["equalize_events_method"] = """
+method : ``'truncate'`` | ``'mintime'`` | ``'random'``
+    If ``'truncate'``, events will be truncated from the end of each event
+    list. If ``'mintime'``, timing differences between each event list will be
+    minimized. If ``'random'``, events will be randomly selected from each event
+    list.
```
Comment (larsoner) [inline mne/utils/docs.py:1297]:
```suggestion
    list.
    
    .. versionadded:: 1.8
```

**Review by larsoner — APPROVED** [review]
Merged `main` which should fix the dep issue and pushed a `versionadded`, marking for merge-when-green thanks in advance @mscheltienne !

**Comment by mscheltienne** [comment]
@vinzrochas FYI

**Pull request merged**